manifests/init.pp

```diff
@@ -12,7 +12,7 @@
   ## IF IN FUTURE WE SET PARAMETERS
   #ensure_resource( 'class', '::apache::mod::ldap', lookup('apache::mod::ldap') )
   #include ::apache::mod::proxy
-  ensure_resource( 'class', '::apache::mod::proxy', lookup('apache::mod::proxy') )
+  #ensure_resource( 'class', '::apache::mod::proxy', lookup('apache::mod::proxy') )
```

Comment:
This needs to not be commented out. Many of our websites make use of apache's mod_proxy (e.g. jira, metrics, wiki, etc).

Perhaps we just check to see if `apache::mod::proxy` is defined and only ensure if it is defined:
```suggestion
    if ( lookup('apache::mod::proxy') ) {
      ensure_resource( 'class', '::apache::mod::proxy', lookup('apache::mod::proxy') )
    }
```
I'm not sure that is quite the correct condition to check for, but something like that.

Reply:
What's the difference between `ensure_resource( 'class', '::apache::mod::proxy', lookup('apache::mod::proxy')` and `include ::apache::mod::proxy`?
Delta has been stuck on v0.1.1 for a while, which doesn't have either of these lines, and I have been writing the include line in profiles.

Reply:
My memory is that simply including `include ::apache::mod::proxy` was not letting use set necessary configuration for `apache::mod::proxy`. So this was done as a work around to get it to work.

But when I search asd-pup-control (e.g. https://git.ncsa.illinois.edu/search?search=apache%3A%3Amod%3A%3Aproxy&nav_source=navbar&project_id=589&group_id=401&search_code=true&repository_ref=main) I don't see where we are actually configuring anything anywhere.

We just set the following everywhere:
```
apache::mod::proxy: {}
```

So I'm not really sure.

Perhaps it would be better to switch to adding and using the following variables?
- `profile_website::proxy_enable`
- `profile_website::proxy_options`

Reply:
The quick fix is to add the following to your `data/common.yaml` file in your control repo:
```
apache::mod::proxy: {}
```

But I agree this should be fixed longer term to be more intuitive and useful. I just caution that it will need to be thoroughly tested everywhere (mostly with asd-control-repo hosts).

Reply:
> My memory is that simply including `include ::apache::mod::proxy` was not letting use set necessary configuration for `apache::mod::proxy`. So this was done as a work around to get it to work.
> 
> But when I search asd-pup-control (e.g. https://git.ncsa.illinois.edu/search?search=apache%3A%3Amod%3A%3Aproxy&nav_source=navbar&project_id=589&group_id=401&search_code=true&repository_ref=main) I don't see where we are actually configuring anything anywhere.
> 
> We just set the following everywhere:
> 
> ```
> apache::mod::proxy: {}
> ```
> 
> So I'm not really sure.
> 
> Perhaps it would be better to switch to adding and using the following variables?
> 
> * `profile_website::proxy_enable`
> * `profile_website::proxy_options`

There is actually an upstream change as well. I used to have
```
apache::mod::proxy:
  allow_from: 'all'
```
But now they have made that parameter an Address type and chose to use an Undef value in place of 'all'. So I too no longer have anything to configure for it.